packages/vertexai/src/backend.test.ts

```diff
@@ -0,0 +1,35 @@
+import { expect } from "chai";
+import { GoogleAIBackend, VertexAIBackend } from "./backend";
+import { BackendType } from "./public-types";
+import { DEFAULT_LOCATION } from "./constants";
+
+describe('Backend', () => {
+  describe('GoogleAIBackend', () => {
+  it('sets backendType to GOOGLE_AI', () => {
+      const backend = new GoogleAIBackend();
+      expect(backend.backendType).to.equal(BackendType.GOOGLE_AI);
+    });
+  });
+  describe('VertexAIBackend', () => {
+    it('set backendType to VERTEX_AI', () => {
+      const backend = new VertexAIBackend();
+      expect(backend.backendType).to.equal(BackendType.VERTEX_AI);
+      expect(backend.location).to.equal(DEFAULT_LOCATION);
+    });
+    it('sets custom location', () => {
+      const backend = new VertexAIBackend('test-location');
+      expect(backend.backendType).to.equal(BackendType.VERTEX_AI);
+      expect(backend.location).to.equal('test-location');
+    });
+    it('sets custom location even if empty string', () => {
```

Comment:
Why do we want that?

Reply:
We currently don't do any validation on location passed, so users can pass things like `' '` , `'x'`, etc without getting errors in the SDK or falling back to the default. I believe allowing `''` fits within this approach.

Do you think `''` should be treated the same as `null` or `undefined` and fallback to the default?

Reply:
I guess it doesn't really matter either way but I do think it's weird to treat null and empty string differently as both of them have to be put there deliberately and with the empty string we're assuming someone putting it there deliberately does not want the default, while with null, we're assuming they do? If you want null to cause it to be default I think anything falsy should also cause default (so maybe `if (!location)`)